### PR TITLE
makesrpm: add initial support for patch guards

### PR DESCRIPTION
**Pull request by srowe**
Add a guard parameter to parse_patchseries() and use it to select
guarded patches. No caller defines it yet so it causes no positive
guards and all negative guards to be applied.

Signed-off-by: Simon Rowe <simon.rowe@eu.citrix.com>

### CODE REVIEW

**Comment by srowe** [comment]
Jira ticket is CA-215582

File: planex/makesrpm.py [hunk 1]
```diff
@@ -33,16 +34,27 @@ def parse_args_or_exit(argv):
     return parser.parse_known_args(argv)
 
 
-def parse_patchseries(series_file):
+def parse_patchseries(series_file, guard = ''):
     """
     Parse series file and return the list of patches
     """
+    guard_re = re.compile(r'([\S]+)( #.*)?')
```
Comment (lindig) [inline planex/makesrpm.py]: Would it be better to use `\s` instead of a space before `#`?

Comment (srowe) [inline planex/makesrpm.py]: Possibly, I'm struggling to find a formal definition of the patch guard format. The convention is a single space but handling any whitespace would be prudent.

**Comment by euanh** [comment]
@srowe: Could you please fix the pylint/pep8 errors?

**Comment by srowe** [comment]
There are still functional issues with this. Let me try again.

File: planex/makesrpm.py [hunk 2]
```diff
@@ -33,16 +34,35 @@ def parse_args_or_exit(argv):
...
+            line = line.strip()
+            if line and not line.startswith('#'):
+                match = guard_re.match(line)
+                if match.group(2):
+                    # guard present
+                    gtype = match.group(2)[2]
+                    guard_patch = match.group(2)[3:]
+
+                    if gtype == '+':
+                        if guard == guard_patch:
```
Comment (euanh) [inline planex/makesrpm.py]: Could you please collapse this inner if into the outer one?

File: planex/makesrpm.py [hunk 3]
```diff
@@ -33,16 +34,35 @@ def parse_args_or_exit(argv):
...
+                            # positive guard matches
+                            patches.append(match.group(1))
+                    elif gtype == '-':
+                        if guard != guard_patch:
```
Comment (euanh) [inline planex/makesrpm.py]: Could you please collapse this inner if into the outer one?

**Comment by euanh** [comment]
I see.   Could you please add a unit test for this function?   To make that easier it might be best to re-cast the function as a filter taking an iterable and returning another one.

I would still prefer to flatten the structure of the logic, something like this:
```
for line in series:
   if not line or line.startswith('#'):
      continue

   match = guard_re.match(line)
   if match.group(2):
      gtype = match.group(2)[2]
      guard_patch = match.group(2)[3:]

      if gtype == '+' and guard != guard.patch:
         continue
      if gtype == '-' and guard == guard.patch:
         continue

   yield match.group(1)
```
